packages/google-pubsub-microservice/src/GooglePubSubServer.ts

```diff
@@ -109,7 +135,22 @@ export class GCPubSubServer extends Server implements CustomTransportStrategy {
         return;
       }
 
-      await handler(message);
+      if (this.shuttingDown) {
+        message.nack();
+
+        return;
+      }
+
+      ++this.counterMessage;
+
+      try {
+        await handler(message);
+      } catch (error) {
+        this.logger.error(`Error handling message from ${subscriptionName}`, error);
+        message.nack();
```

Comment:
This changes the behavior of the library. A try-finally would be enough here, IMHO. Do you agree, @amarlankri?